### DISCUSSION TITLE
How to get a handle to the window created by dioxus::desktop::launch(app);

### DISCUSSION THREAD

**Discussion by Solaris235** [DioxusLabs/dioxus#635] [Q&A] [answered]
I would like to set the initial size of a window, which I see can be done with tao::window::Window.set_inner_size(). and access all the Window methods.
How can I get the handle or pointer to the window. So far all I can get is the DesktopContext which doesn't seem to only have access to some of those methods. At least not that I have found. Am I missing something or do I have to go the long way around and use tao::window.new() and develop the app that way. Has this just no been implemented yet?
Any comments or help would be appreciated. Dioxus looks like it has lots of potential.
Thanks.

**Comment by InfamousVague** [comment, marked as answer]
I'm not sure if this is the preferred way to do things, but it's how we did it, hopefully it helps: https://github.com/Satellite-im/Uplink/blob/dev/src/main.rs#L181-L209

```rs
  use tao::window::WindowBuilder;
  ...

    let window = WindowBuilder::new()
        .with_title(DEFAULT_WINDOW_NAME.read().clone())
        .with_resizable(true)
        .with_inner_size(LogicalSize::new(950.0, 600.0))
        .with_min_inner_size(LogicalSize::new(330.0, 500.0));

    dioxus::desktop::launch_with_props(
        App,
        None,
        |c| c.with_window(|_| window),
    );

}
```

The menu stuff is something some systems (like MacOS) require in order to enable keybinds like copy paste and select all.

**Reply by Solaris235** [reply]
Thanks very much that's what I've been looking for.
I did have to replace the `None` properties argument with` () ` for it to compile.
I was looking for a way to resize and relocate the window once it was created and displayed, possibly by the click of a button.
This is a great help thanks again for the quick response.

**Reply by koskeller** [reply]
`tao` is reexported by `dioxis::desktop`, so you can use it from here:
``` rust
use dioxus::desktop::tao;

fn main() {
    let window = tao::window::WindowBuilder::new().with_resizable(true);
    dioxus::desktop::launch_cfg(app, |c| c.with_window(|_| window));
}
```

**Reply by miaomiaowu0428** [reply]
> `tao`由 重新导出，因此您可以在此处使用它：`dioxis::desktop`
> 
> ```rust
> use dioxus::desktop::tao;
> 
> fn main() {
>     let window = tao::window::WindowBuilder::new().with_resizable(true);
>     dioxus::desktop::launch_cfg(app, |c| c.with_window(|_| window));
> }
> ```

hi
`launch_cfg` or `launch_with_props` is not available in version 0.5, what should I use now?

**Reply by ealmloff** [reply]
You can use the launch builder with a config:
```rust
use dioxus::desktop::tao;

fn main() {
    let window = tao::window::WindowBuilder::new().with_resizable(true);
    dioxus::LaunchBuilder::new().with_cfg(dioxus::desktop::Config::new().with_window(window)).launch(app);
}
```
